src/app/pages/AccountDetailsPage/AccountTransactionsCard.tsx

```diff
@@ -9,15 +9,14 @@ import { ErrorBoundary } from '../../components/ErrorBoundary'
 import { LinkableDiv } from '../../components/PageLayout/LinkableDiv'
 import { CardEmptyState } from './CardEmptyState'
 import { useAccountTransactions } from './hook'
-import { useRequiredScopeParam } from '../../hooks/useScopeParam'
-import { useLoaderData } from 'react-router-dom'
+import { AccountDetailsContext } from './index'
+import { useOutletContext } from 'react-router-dom'
 
 export const accountTransactionsContainerId = 'transactions'
 
 export const AccountTransactionsCard: FC = () => {
+  const { scope, address } = useOutletContext<AccountDetailsContext>()
```

Comment:
My thinking was that this is part of hierarchical navigation; the outer part (the page) is containing the inner part (the tabs). Therefore if the outer part already go the data from the router, it's only fair that it passes it down to it's children. Doing this, we will be free to move the page wherever we want in the navigation tree, and the only thing we have to update is the top part.

   * * *

On the other hand, one could also make the argument that since the tabs are references directly inside the routing tree definition, these _are_ routing level components, on equal level as other page-level components, so they can talk to the router directly.

I'm happy to undo the outlet-based changes.

Reply:
Lemme try out one more version

Reply:
@lukaw3d  How about now? This time, all cards (both those that are embedded on the page + those that are used in tab menus) take props the same way, normally. The props are passed down from the menu, and received in the routes definition.

I think the actual components are very easy to read now. They don't have to interact with router, outlet, or anything. All the tricky parts are confined into the router itself, when we are already concentrating on building a menu.